concrete/codegen/solgen/solgen.go

```diff
@@ -115,12 +116,19 @@ func generateSolidityLibrary(ABI abi.ABI, cABI customABI, config Config) (string
 		"ImportPaths": importPaths,
 	}
 
-	for mIdx, method := range ABI.Methods {
+	methodNames := make([]string, len(ABI.Methods))
```

Comment:
You are initializing a slice of length `len(ABI.Methods)` then appending `len(ABI.Methods)` elements to it. This results in a slice of length `2*len(ABI.Methods)`, half if wich are empty strings. This leads to unintended behavior, as you can see in the invalid methods this commit adds to `CounterPrecompile.sol`.

I'm guessing you meant to initialize a slice of length 0 with capacity `len(ABI.Methods)`. Which you can do with `methodNames := make([]string, 0, len(ABI.Methods))`. The first number is the length, the second is the capacity.

Reply:
My bad! That is exactly what I intended, sorry for the typo.